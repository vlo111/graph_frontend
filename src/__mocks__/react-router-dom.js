import React from 'react';

const rrd = require('react-router-dom');
// Just render plain root element with its children
rrd.BrowserRouter = ({ children }) => <div>{children}</div>;
module.exports = rrd;
