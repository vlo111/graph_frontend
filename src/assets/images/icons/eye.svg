<svg xmlns="http://www.w3.org/2000/svg" width="15" height="8.94" viewBox="0 0 15 8.94">
    <path
            d="M7.5,74.667A8.118,8.118,0,0,0,0,79.137a8.118,8.118,0,0,0,7.5,4.47,8.114,8.114,0,0,0,7.5-4.47A8.114,8.114,0,0,0,7.5,74.667Z"
            transform="translate(0 -74.667)"/>
</svg>
