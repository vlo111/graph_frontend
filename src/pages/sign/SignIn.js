import React, { Component } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { Link, Redirect } from 'react-router-dom';
import { toast } from 'react-toastify';
import { connect } from 'react-redux';
import { ReactComponent as LogoSvg } from '../../assets/images/logo.svg';
import fbImg from '../../assets/images/icons/fb.svg';
import googleImg from '../../assets/images/icons/google.png';
import { signInRequest } from '../../store/actions/account';
import WrapperSign from '../../components/WrapperSign';
import Input from '../../components/form/Input';
import Button from '../../components/form/Button';
import OAuthButtons from "../../components/account/OAuthButtons";

class Login extends Component {
  static propTypes = {
    signInRequest: PropTypes.func.isRequired,
    token: PropTypes.string.isRequired,
  }

  constructor(props) {
    super(props);
    this.state = {
      loading: false,
      requestData: {
        email: '',
        password: '',
      },
    };
  }

  handleTextChange = (value, path) => {
    const { requestData } = this.state;
    _.set(requestData, path, value);
    this.setState({ requestData });
  }

  signIn = async (ev) => {
    ev.preventDefault();
    const { requestData } = this.state;
    this.setState({ loading: true });
    const { payload } = await this.props.signInRequest(requestData.email, requestData.password);
    this.setState({ loading: false });
    const { data = {} } = payload;
    if (data.status !== 'ok') {
      toast.dismiss(this.toast);
      this.toast = toast.error('Invalid email or password');
    }
  }

  render() {
    const { token } = this.props;
    const { requestData } = this.state;
    if (token) {
      return (<Redirect to="/" />);
    }
    return (
      <WrapperSign>
        <div className="left signIn">
          <Link to="/">
            <LogoSvg className="logo white" />
          </Link>
        </div>
        <div className="right">
          <div>
            <form onSubmit={this.signIn} id="login" className="authForm">
              <h1>Sign in to</h1>
              <LogoSvg className="logo orange" />
              <div className="socialLogin">
                <h4>Sign in using</h4>
                <OAuthButtons />
              </div>

              <div className="hr">or</div>
              <Input
                name="email"
                type="email"
                label="Email address"
                value={requestData.email}
                onChangeText={this.handleTextChange}
              />
              <Input
                name="password"
                type="password"
                label="Password"
                value={requestData.password}
                onChangeText={this.handleTextChange}
              />
              <Link to="/sign/forgot-password" className="forgotPassword">Forgot password?</Link>

              <Button type="submit" className="submit" color="orange">
                Sign In
              </Button>
            </form>
            <p className="switchSignMode">
              {"Don't have an admin yet? "}
              <Link to="/sign/sign-up">Get started</Link>
            </p>
          </div>
        </div>

      </WrapperSign>
    );
  }
}

const mapStateToProps = (state) => ({
  token: state.account.token,
});

const mapDispatchToProps = {
  signInRequest,
};

const Container = connect(
  mapStateToProps,
  mapDispatchToProps,
)(Login);

export default Container;
