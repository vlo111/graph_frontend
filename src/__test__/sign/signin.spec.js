import React from 'react';
import { Route, Switch } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { fireEvent } from '@testing-library/react';
import SignIn from '../../pages/sign/SignIn';
import renderWithReduxNRouter from '../render';
import '@testing-library/jest-dom/extend-expect';
import { ReactComponent as EyeSvg } from '../../assets/images/icons/eye.svg';
import { ReactComponent as LogoSvg } from '../../assets/images/araks_logo.svg';

const RouterComponent = () => (
  <>
    <Switch>
      <Route path="/sign/sign-in" component={SignIn} />
    </Switch>
  </>
);

/**
 * Mock google api
 * Initial google api for get access token
 */
// ------------
// ------------

/**
 * test social media api in the page
 * check social media buttons
 */
describe('check social media action', () => {
  it('check all logins social networks', () => {
    const { getByTestId } = renderWithReduxNRouter(<RouterComponent />,
      {
        route: '/sign/sign-in',
      });

    /**
     * get token after click google from social media
     */
    fireEvent.click(getByTestId('google'));
  });
});

/**
 * snapshots sign in elements
 * check social media block
 * check logo with form fields
 */
describe('html elements displayed correctly', () => {
  it('should be match sign in button', () => {
    /**
     * match sign in button
     * @type {*|{children: *, type: *, props: {}|{}}}
     */
    const signinBtn = renderer.create(
      <button className="ghButton submit orange alt">Sign In</button>,
    ).toJSON();
    expect(signinBtn).toMatchSnapshot();

    /**
     * match email input
     * @type {*|{children: *, type: *, props: {}|{}}}
     */
    const email = renderer.create(
      <div className="ghFormField ghInput">
        <input placeholder="Email address" />
      </div>,
    ).toJSON();
    expect(email).toMatchSnapshot();

    /**
     * match password input
     * @type {*|{children: *, type: *, props: {}|{}}}
     */
    const password = renderer.create(
      <div className="ghFormFieldPassword   ghFormField ghInput">
        <span className="icon ">
          <EyeSvg
            width={14}
          />
        </span>
        <input placeholder="Password" type="password" />
      </div>,
    ).toJSON();

    expect(password).toMatchSnapshot();

    /**
     * match forget password field
     * @type {*|{children: *, type: *, props: {}|{}}}
     */
    const forget = renderer.create(
      <a className="forgotPassword">Forgot password?</a>,
    ).toJSON();

    expect(forget).toMatchSnapshot();

    /**
     * match logo
     * @type {*|{children: *, type: *, props: {}|{}}}
     */
    const logo = renderer.create(
      <LogoSvg className="logo white" />,
    ).toJSON();

    expect(logo).toMatchSnapshot();
  });
});

/**
 * test elements exist in sign_in page
 * check text dom elements with buttons
 * check social media buttons
 */
describe('check sign in dom elements', () => {
  it('should be the text elements', () => {
    const { getByTestId, getByText } = renderWithReduxNRouter(<RouterComponent />,
      {
        route: '/sign/sign-in',
      });

    /**
     * should be the forget password text
     */
    expect(getByText('Don\'t have an admin yet?')).toBeInTheDocument();

    /**
     * should be sign in using social media header
     */
    expect(getByText('Sign in using')).toBeInTheDocument();

    /**
     * should be email input field
     */
    expect(getByTestId('email')).toBeInTheDocument();

    /**
     * should be password input field
     */
    expect(getByTestId('password')).toBeInTheDocument();

    /**
     * should be facebook field
     */
    expect(getByTestId('facebook')).toBeInTheDocument();

    /**
     * should be linked In field
     */
    expect(getByTestId('linkedin')).toBeInTheDocument();

    /**
     * should be google field
     */
    expect(getByTestId('google')).toBeInTheDocument();

    /**
     * should be twitter field
     */
    expect(getByTestId('twitter')).toBeInTheDocument();
  });
});
