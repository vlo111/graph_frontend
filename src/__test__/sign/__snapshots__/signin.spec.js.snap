// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`html elements displayed correctly should be match sign in button 1`] = `
<button
  className="ghButton submit orange alt"
>
  Sign In
</button>
`;

exports[`html elements displayed correctly should be match sign in button 2`] = `
<div
  className="ghFormField ghInput"
>
  <input
    placeholder="Email address"
  />
</div>
`;

exports[`html elements displayed correctly should be match sign in button 3`] = `
<div
  className="ghFormFieldPassword   ghFormField ghInput"
>
  <span
    className="icon "
  >
    <svg
      width={14}
    >
      eye.svg
    </svg>
  </span>
  <input
    placeholder="Password"
    type="password"
  />
</div>
`;

exports[`html elements displayed correctly should be match sign in button 4`] = `
<a
  className="forgotPassword"
>
  Forgot password?
</a>
`;

exports[`html elements displayed correctly should be match sign in button 5`] = `
<svg
  className="logo white"
>
  araks_logo.svg
</svg>
`;
